jwql/website/apps/jwql/templates/base.html

```diff
@@ -75,7 +78,26 @@
 	        	</ul>
 	        </div>
 
-	    	<a class="navbar-right" href="https://github.com/spacetelescope/jwql" target="_blank"><img src={{ static('img/githublogo.svg') }} height=25 width=25></a>
+	        <!-- Oauth login and GitHub -->
+	        <div class="navbar-right">
+	            <ul class="navbar-nav mr-auto">
+	            	{% if user.ezid %}
+	            		<li>
+	            			<a class="nav-link" href="https://auth.mastdev.stsci.edu/tokens" id="oauth_user" style="transform:translateY(20%);">{{ user.ezid }}</a>
```

Comment:
So, I am still getting a `UndefinedError: 'user' is undefined` for what seems to be every page except the homepage. I think this is because the `user` variable, which is used here in `base.html` and thus is loaded on every webpage, isn't always getting passed into the `context` dictionary (like it is on line ~298 of `views.py`). 

I imagine adding `user` as an argument to all the views and as an entry in each view's context dict might fix this?

Reply:
I tried to do this for the `instruments()` view and it is unfortunately not quite as simple as I would have hoped. It seems like Django passing specific variables into the view (e.g. `inst`) messes up the `auth_info` decorator. Hopefully you might have better luck with it? Or an idea for another kind of fix?

Reply:
@laurenmarietta Good catch, I was so fixated on getting this working on the homepage that I neglected to notice that it broke all of the other pages 😱 .  Yes, I think the problem is just as you describe, I will try to find a solution.

Reply:
@laurenmarietta I think I have come up with a solution for this (f73df96) -- it was a matter of passing `user` to each view, and doing some tricky stuff with `**kwargs` in the `auth_info` decorator.

Though this appears to be working, i'm not 100% convinced that this is the best way to do it.  Perhaps an issue to tackle in a future issue/PR.  I think there may exist a better way in separating `context` needed for `base` from `context` needed for the other templates.  At the very least, perhaps we could define a convenient `get_base_context()` function that can be called in each view.

Reply:
A lovely fix - all the pages are working now. Thank you!

I definitely don't have any intuition as to whether it is normal to pass the user as an argument into all the views... As per usual, I'd say the Django docs are a good resource for how to do things sleek-ly. I have just barely skimmed it, but it seems like Django might have a way to let us integrate our customized MAST authentication into their usual authentication system? For a future PR: https://docs.djangoproject.com/en/2.1/topics/auth/

Reply:
@laurenmarietta Thanks for looking into that.  I will open a separate "make passing base context better" issue for this.